src/WebJobs.Script.WebHost/WebJobsScriptHostService.cs

```diff
@@ -130,7 +131,19 @@ private IHost ActiveHost
 
         public IServiceProvider Services => ActiveHost?.Services;
 
-        public ScriptHostState State { get; private set; }
+        public ScriptHostState State
+        {
+            get
+            {
+                return _state;
+            }
+
+            private set
+            {
+                _state = value;
+                _logger?.HostStateChanged(_state);
```

Comment:
don't need the null check on _logger - it's set in ctor

Reply:
Also, recommend only writing the "state changed" log if it actually changed. Compare new value to existing first.